core/validatorapi/router.go

```diff
@@ -1039,46 +1034,59 @@ func getValidatorIDs(query url.Values) []string {
 	return resp
 }
 
-// getValidatorByID returns the validator and true with id being either a pubkey or a validator index.
-// It returns false if the validator is not found.
-func getValidatorByID(ctx context.Context, p eth2client.ValidatorsProvider, stateID, id string) (*eth2v1.Validator, bool, error) {
-	if strings.HasPrefix(id, "0x") {
-		pubkey, err := tblsconv.KeyFromCore(core.PubKey(id))
-		if err != nil {
-			return nil, false, errors.Wrap(err, "decode public key hex")
-		}
-		eth2Pubkey, err := tblsconv.KeyToETH2(pubkey)
-		if err != nil {
-			return nil, false, err
+// getValidatorsByID returns the validators with ids being either pubkeys or validator indexes.
+func getValidatorsByID(ctx context.Context, p eth2client.ValidatorsProvider, stateID string, ids ...string) ([]v1Validator, error) {
+	if len(ids) == 0 {
+		return nil, errors.New("no validator ids provided")
+	}
+
+	flatten := func(kvs map[eth2p0.ValidatorIndex]*eth2v1.Validator) []v1Validator {
+		var vals []v1Validator
+		for _, v := range kvs {
+			vals = append(vals, v1Validator(*v))
 		}
 
-		temp, err := p.ValidatorsByPubKey(ctx, stateID, []eth2p0.BLSPubKey{eth2Pubkey})
-		if err != nil {
-			return nil, false, err
+		return vals
+	}
+
+	if strings.HasPrefix(ids[0], "0x") {
```

Comment:
why do we assume that all `ids` are either pubkeys or validator indexes?

Reply:
because it is

Reply:
For ex, ids: ["1", "0xbef...", "0xaac..", "4"]

Reply:
that is not a valid usecase we intend to support

Reply:
i'm thinking in terms of how the endpoint is itself structured - https://ethereum.github.io/beacon-APIs/#/Beacon/getStateValidators.

`
Information will be returned for all indices or public key that match known validators. If an index or public key does not match any known validator, no information will be returned but this will not cause an error. There are no guarantees for the returned data in terms of ordering; both the index and public key are returned for each validator, and can be used to confirm for which inputs a response has been returned.
`

but, it's upto us how we want to structure it on our end

Reply:
we cannot handle a mix, since goeth2 doesn't support a mix